src/opers.h

```diff
@@ -27,78 +63,155 @@ extern Obj TRY_NEXT_METHOD;
+}
+
+static inline void SET_FLAG1_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->flag1 = x;
+}
 
 
 /****************************************************************************
 **
 *F  FLAG2_FILT( <oper> )  . . . . . . . . . .  flag 2 list of an `and' filter
 */
-#define FLAG2_FILT(oper)        (*            (ADDR_OBJ(oper) +17     ) )
+static inline Obj FLAG2_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->flag2;
+}
+
+static inline void SET_FLAG2_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->flag2 = x;
+}
 
 
 /****************************************************************************
 **
 *F  FLAGS_FILT( <oper> )  . . . . . . . . . . . . . . . . . flags of a filter
 */
-#define FLAGS_FILT(oper)        (*            (ADDR_OBJ(oper) +18     ) )
+static inline Obj FLAGS_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->flags;
+}
+
+static inline void SET_FLAGS_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->flags = x;
+}
 
 
 /****************************************************************************
 **
 *F  SETTER_FILT( <oper> ) . . . . . . . . . . . . . . . .  setter of a filter
 */
-#define SETTR_FILT(oper)        (*            (ADDR_OBJ(oper) +19     ) )
+static inline Obj SETTR_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->setter;
+}
+
+static inline void SET_SETTR_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->setter = x;
+}
 
 
 /****************************************************************************
 **
 *F  TESTR_FILT( <oper> )  . . . . . . . . . . . . . . . .  tester of a filter
 */
-#define TESTR_FILT(oper)        (*            (ADDR_OBJ(oper) +20     ) )
+static inline Obj TESTR_FILT(Obj oper)
+{
+    return CONST_OPER(oper)->tester;
+}
+
+static inline void SET_TESTR_FILT(Obj oper, Obj x)
+{
+    OPER(oper)->tester = x;
+}
 
 
 /****************************************************************************
 **
 *F  METHS_OPER( <oper> )  . . . . . . . . . . . . method list of an operation
 */
-#define METHS_OPER(oper,i)      (*            (ADDR_OBJ(oper) +21+ (i)) )
+static inline Obj METHS_OPER(Obj oper, Int i)
+{
+    return CONST_OPER(oper)->methods[i];
+}
+
+static inline void SET_METHS_OPER(Obj oper, Int i, Obj x)
+{
+    OPER(oper)->methods[i] = x;
+}
 
 
 /****************************************************************************
 **
 *F  CACHE_OPER( <oper> )  . . . . . . . . . . . . . . . cache of an operation
 */
-#define CACHE_OPER(oper,i)      (*            (ADDR_OBJ(oper) +29+ (i)) )
+static inline Obj CACHE_OPER(Obj oper, Int i)
+{
+    return CONST_OPER(oper)->cache[i];
+}
+
+static inline void SET_CACHE_OPER(Obj oper, Int i, Obj x)
+{
+    OPER(oper)->cache[i] = x;
+}
+
 
 /****************************************************************************
 **
 *F  ENABLED_ATTR( <oper> ) . . . . true if the operation is an attribute and
 **                                 storing is enabled (default) else false
 */
+static inline Int ENABLED_ATTR(Obj oper)
+{
+    Obj val = CONST_OPER(oper)->enabled;
```

Comment:
Was trying to figure out why we go int -> obj -> int here, instead of just storing an int?

Reply:
The code previously did store 0 and 1 directly. But note that we use `MarkAllSubBags` for `T_FUNCTION`, so strictly speaking, any any data in there should be a valid bag / object. Now, by coincidence, the integers `0` and `1` are fine (the latter being the immediate integer encoding for 0). But I felt it was wrong to do that blindly. 

That said, I could restore the old behavior, and add a comment which points out what is going on; I'd also enforce that only 0 and 1 get stored. (BTW, it's a bit annoying that we waste a full word to essentially store a single bit).

Reply:
The fact we call markAllSubbags is good enough, and I definately wouldn't want to start relying on how we happen to store immediate integers :)